runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/fn/control/ProcessRemoteBundleOperation.java

```diff
@@ -84,10 +170,21 @@ public void start() throws Exception {
 
   @Override
   public void process(Object inputElement) throws Exception {
-    LOG.debug(String.format("Sending value: %s", inputElement));
+    LOG.debug(String.format("Sending element: %s", inputElement));
+    String mainInputPCollectionId = executableStage.getInputPCollection().getId();
+    FnDataReceiver<WindowedValue<?>> mainInputReceiver =
+        remoteBundle.getInputReceivers().get(mainInputPCollectionId);
+
+    // TODO(BEAM-6274): Is this always true? Do we always send the input element to the main input receiver?
```

Comment:
Based on the current implementation of ExecutableStage, it's true that there is only one input PCollection(which is mapping to one InputReceiver). I'm not sure whether it's true in the future, @robertwb do you have any insights on this?

Reply:
This will not be true once timers are implented, and possibly not for SDF or other stage shapes.